src/tile.rs

```diff
@@ -2,26 +2,187 @@ use bytes::Bytes;
 
 use crate::decoder::DecoderRegistry;
 use crate::error::AsyncTiffResult;
-use crate::tiff::tags::{CompressionMethod, PhotometricInterpretation};
+use crate::predictor::RevPredictorRegistry;
+use crate::reader::Endianness;
+use crate::tiff::tags::{
+    CompressionMethod, PhotometricInterpretation, PlanarConfiguration, Predictor, SampleFormat,
+};
 use crate::tiff::{TiffError, TiffUnsupportedError};
 
+/// All info that may be used by a predictor
+///
+/// Most of this is used by the floating point predictor
+/// since that intermixes padding into the decompressed output
+///
+/// Also provides convenience functions
+///
+#[derive(Debug, Clone, Copy)]
+pub struct PredictorInfo<'a> {
```

Comment:
I'm wondering if it makes sense to add this `PredictorInfo` struct because all of the information provided by it is already available on the IFD, and it seems like the main way of creating this struct is by calling `IFD.get_predictor_info`.

Any implementation that is unpredicting a tile will also have access to the IFD, so I think it could make sense to remove this intermediate struct entirely in favor of just interacting with the IFD directly.

Reply:
`Tile` above says in its docstrings that it has all necessary information to decode a tile. I'm fine with removing the intermediate struct, but then I think it'd also make sense to not store `CompressionMethod` on the tile itself, since the decoding operation needs access to the IFD?

Reply:
I actually disagree with @geospatial-jeff here. We created this `Tile` construct because we want to enable some use cases (including some of Jeff's) where the user doesn't actually want to decompress the data from the COG; they want to pass on the compressed data e.g. from a tile server. 

If we want to expose the `Tile` construct to bindings like Python, it can't have a lifetime. In that case, the `Tile` needs to maintain all data necessary to handle decoding, unprediction, and endianness checking. But the full IFD is quite large. So I think having a `PredictorInfo` struct is quite useful so that we can encapsulate only this data that we actually need to put in `Tile`. And it's useful that all of this `PredictorInfo` is also `Copy`.

Reply:
I see, I didn't understand the limitation around "lifetime".  `PredictorInfo` makes sense to me if the goal is to have strong encapsulation around just the parts of the IFD that are needed for decoding.